egret/parsers/rts_gmlc/parser.py

```diff
@@ -0,0 +1,838 @@
+    for idx,row in timeseries_pointer_df.iterrows():
+        # Skip rows we don't ingest
+        if not row['Category'] in params_of_interest:
+            continue
+        if not row['Parameter'] in params_of_interest[row['Category']]:
+            continue
+
+        # Skip generators not in skeleton
+        if row['Category'] == 'Generator' and not row['Object'] in elements['generator']:
+            continue
+        # Skip areas not in skeleton
+        if row['Category'] == 'Area' and not row['Object'] in elements['area']:
+            continue
+
+        is_reserve = (row['Category'] == 'Reserve')
+        if is_reserve:
+            # Skip unrecognized reserve names
+            name = str(row['Object'])
+            if not is_valid_reserve_name(name, model_data):
+                continue
+
+        # Read the timeseries file if we haven't already, using the
+        # canonical file path as a key into previously read filenames.
+        fname = os.path.abspath(os.path.join(rts_gmlc_dir, row['Data File']))
+        if not fname in timeseries_file_map:
+            sim = row['Simulation']
+            data = _read_timeseries_file(fname, minutes_per_period[sim], 
+                                         start_time, end_time, row['Object'])
+            timeseries_file_map[fname] = data
+
+        # Save a reference to the relevant data as a Series
+        timeseries_pointer_df.at[idx,'Series'] = timeseries_file_map[fname][row['Object']]
+
+    # Remove columns that we don't want to preserve
+    keepers= {'Simulation', 'Category', 'Object', 'Parameter', 'Series'}
+    for c in timeseries_pointer_df.columns:
+        if not c in keepers:
+            timeseries_pointer_df.pop(c)
+
+    # Remove irrelevant rows
+    timeseries_pointer_df.dropna(subset=['Series'], inplace=True)
+
+    # Sort by simulation
+    timeseries_pointer_df.sort_values(by='Simulation', inplace=True)
+
+    return timeseries_pointer_df
+
+def _parse_datetimes_if_strings(begin_time:Union[datetime,str], end_time:Union[datetime,str]):
+    '''
+    Ensure both dates are datetimes, parsing date strings if necessary.
+
+    Returns
+    -------
+    begin_time:datetime
+        The begin_time as a datetime, parsing it if necessary
+    end_time:datetime
+        The end_time as a datetime, parsing it if necessary
+    '''
+
+    datetime_format = "%Y-%m-%d %H:%M:%S"
+
+    datestr = "YYYY-DD-MM"
+    midnight = " 00:00:00"
+
+    if isinstance(begin_time,datetime):
+        pass
+    elif isinstance(begin_time,str):
+        if len(begin_time) == len(datestr):
+            begin_time += midnight
+        begin_time = datetime.strptime(begin_time,datetime_format)
+    else:
+        raise ValueError("Unable to parse begin_time")
+
+    if isinstance(end_time,datetime):
+        pass
+    elif isinstance(end_time,str):
+        if len(end_time) == len(datestr):
+            end_time += midnight
+        end_time = datetime.strptime(end_time,datetime_format)

```

Comment:
Similar to above (probably want a function for this repeated code).

Reply:
I created a function with the repeated code. I also switched from strptime with a fixed format to dateutil.parser.parse, which infers the date format from the string. See _convert_to_datetime().